### ISSUE TITLE
Keyboard accessibility bug in global search

### ISSUE THREAD

**Issue by zachlysobey** [backstage/backstage#13680]
## Expected Behavior

It should be possible to use the global search without use of a pointer device (mouse).

## Actual Behavior

When using the search, if a search result is focused, hitting `enter` or `space` does not navigate to the selected result. It just closes the search dialog.

## Steps to Reproduce

1. Open the global search dialog
2. Type a search term that yields results
3. Tab into one of the results
4. Hit `enter` or `tab`

## Context

Accessibility is important ;)

## Your Environment

*reproducible on `demo.backstage.io`*

**Comment by zachlysobey** [comment]
As an update, its been pointed out to me there is a workaround, and that if you tab a second time after a search result is focused, the same result remains focused (or probably a different HTML element) and *that one* behaves correctly with `enter` or `space`

**Comment by garyniemen** [comment]
@emmaindal What do you think about this one?

**Comment by emmaindal** [comment]
Hello @zachlysobey thanks for opening this issue, I agree - we should definitely have better accessibility support in place. Is this something you want to work on? Would be a suuper valuable contribution. I'll add the "help wanted" label on this issue too in case someone else have time to pick it up!
